lib/sshkit/ssh/channel.ex

```diff
@@ -39,6 +39,22 @@ defmodule SSHKit.SSH.Channel do
     end
   end
 
+  @doc """
+  Activates a subsystem on a channel.
+
+  Returns `:success`, `:failure` or `{:error, reason}`.
+
+  For more details, see [`:ssh_connection.subsystem/4`](http://erlang.org/doc/man/ssh_connection.html#subsystem-4).
+  """
+  @spec subsystem(channel :: struct(), subsystem :: String.t(), options :: list()) ::
+          :success | :failure | {:error, reason :: String.t()}
+  def subsystem(channel, subsystem, options \\ []) do
+    timeout = Keyword.get(options, :timeout, :infinity)
+    impl = Keyword.get(options, :impl, :ssh_connection)
+
+    impl.subsystem(channel.connection.ref, channel.id, to_charlist(subsystem), timeout)
+  end
```

Comment:
Nice 👍

Maybe you could just move it down below the `build/3` which is used in `open/2` above so that the two are not separated? 🤗

Reply:
whoops, yeah that makes a lot of sense, changed it!